File: content/2013/go4python/fib.go
// +build OMIT

package main

import "fmt"

func fib(n int) int {
	a, b := 0, 1
	for i := 0; i < n; i++ {
		a, b = b, a+b
	}
	return b
}

func fibRec(n int) int {
	if n <= 1 {
		return 1
	}
	return fibRec(n-1) + fibRec(n-2)
}

func main() {
	for i := 0; i < 10; i++ {
		fmt.Println(fib(i), fibRec(i))
	}
}

File: content/2013/go4python/fib.py
#!/usr/bin/python

def fib(n):
    a, b = 0, 1
    for i in range(n):
        a, b = b, a + b
    return b

def fib_rec(n):
	if n <= 1:
		return 1
	else:
		return fib_rec(n-1) + fib_rec(n-2)

for x in range(10):
	print fib(x), fib_rec(x)
